File: src/main/java/me/ag2s/epublib/util/commons/io/IOConsumer.java
package me.ag2s.epublib.util.commons.io;

import java.io.IOException;
import java.util.Objects;

@FunctionalInterface
public interface IOConsumer {
   void accept(Object t) throws IOException;

   default IOConsumer andThen(final IOConsumer after) {
      Objects.requireNonNull(after);
      return (t) -> {
         this.accept(t);
         after.accept(t);
      };
   }
}

File: src/main/kotlin/me/ag2s/epublib/util/commons/io/IOConsumer.kt
package me.ag2s.epublib.util.commons.io

import java.io.IOException

fun interface IOConsumer<T> {
    @Throws(IOException::class)
    fun accept(value: T)

    fun andThen(after: IOConsumer<T>): IOConsumer<T> = IOConsumer { value -> accept(value); after.accept(value) }
}
